openmp/runtime/src/kmp_lock.h

```diff
@@ -120,8 +120,15 @@ extern void __kmp_validate_locks(void);
 
 struct kmp_base_tas_lock {
   // KMP_LOCK_FREE(tas) => unlocked; locked: (gtid+1) of owning thread
+#if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__ && KMP_ARCH_PPC64
```

Comment:
I question whether ```KMP_ARCH_PPC64``` should be there.

Reply:
I thought to use `KMP_ARCH_PPC64` to limit the scope. This issue may also affect s390x which I believe is also big-endian. I've changed to guard the order flipping with `#if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__ && __LP64__`. @iii-i please let me know if it does not work for s390x. The reason for guarding it with `__LP64__` is the original ordering works as designed in 32-bit mode both little- and big-endian. If these members are flipped for 32-bit big-endian, tags of a direct lock will take the second word while the address of the entry allocated from the indirect lock table will take the first word.

Reply:
> I thought to use `KMP_ARCH_PPC64` to limit the scope. This issue may also affect s390x which I believe is also big-endian.

Ya, I was also thinking about other big-endian processors that might be affected like mips64 or if sparc support is eventually added sparc64.